### PR TITLE
Issue #3132 - Improve documentation for the jetty server dump

### LINKED ISSUE #3132: update dump-tool documentation
The [jetty-dump-tool](https://www.eclipse.org/jetty/documentation/current/jetty-dump-tool.html) documentation needs to be updated:
- I'm not sure calling it a tool is the right thing, as it implies a separate executable.  I think "jetty dump feature" is better.
- The [configuring-dump-feature](https://www.eclipse.org/jetty/documentation/current/jetty-dump-tool.html#configuring-dump-feature) is written as how to modify `jetty.xml`.   Instead is should be written in terms of the `jetty.server.dumpAfterStart` and `jetty.server.dumpBeforeStop` properties, which can be set either on the command line or in the `server.ini` file.   (As an aside I think the [How to Configure Jetty](https://www.eclipse.org/jetty/documentation/current/quick-start-configure.html#quickstart-config-how) could do with an overview that says something along the lines of: for the most part jetty is configured by enabling modules and setting parameters; modules can be enabled either on the command line (example) or in an ini file which can be created with `--add-to-start`;  parameters may be set on the command line or by editing an ini file. )
- We don't need the [extra-threadpool-info](https://www.eclipse.org/jetty/documentation/current/jetty-dump-tool.html#extra-threadpool-info) section.  This should just be describing another property in the section above: `jetty.threadPool.detailedDump` which also can be set on the command line or in the `threadpool.ini` file.
- The examining section is just too big and is also out of date.    It should just have a very smaller example, plus a detailed explanation of the dump key

### PR DESCRIPTION
**Pull request by lachlan-roberts**
closes #3132

### CODE REVIEW

File: jetty-documentation/src/main/asciidoc/administration/logging/jetty-server-dump.adoc [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+[[jetty-server-dump]]
+=== Jetty Server Dump
+
+The dump feature in Jetty provides a good snapshot of the status of the threadpool, select sets, classloaders, and so forth.
```
Comment (gregw) [inline jetty-documentation/src/main/asciidoc/administration/logging/jetty-server-dump.adoc]:
```suggestion
The dump feature in Jetty provides a snapshot of the bean containment tree of the main Jetty components together with a summary of their configuration. This includes threadpool, connectors,  contexts, webapplications, servlets and so forth. 
```

File: jetty-documentation/src/main/asciidoc/administration/logging/jetty-server-dump.adoc [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+[[jetty-server-dump]]
+=== Jetty Server Dump
+
+The dump feature in Jetty provides a good snapshot of the status of the threadpool, select sets, classloaders, and so forth. 
+To get maximum detail from the dump, you need to `setDetailDump(true)` on any `QueuedThreadPools` you are using. 
+You can do this by a direct call if you are embedding Jetty, or by using properties for standalone Jetty.
```
Comment (gregw) [inline jetty-documentation/src/main/asciidoc/administration/logging/jetty-server-dump.adoc]: This is a bit too detailed to be in the introduction.   It should be in the section below.  Perhaps say what extra detail is provided and why you might want it turned off (verbosity)

File: jetty-documentation/src/main/asciidoc/administration/logging/jetty-server-dump.adoc [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+The `dump` method is on the Server instance and many of its nested components (Handlers, Connectors, and so forth). 
+Dumps may be obtained by calling these methods either in code or via JMX (see xref:using-jmx[]).
+
+The Server MBean has a `dump()` method, which dumps everything, plus a `dumpStdErr()` operation that dumps to StdErr rather than replying to JConsole.
+
+[[examing-jetty-distro-dump]]
+==== Explanation of the Dump Key
+
+- +- bean is a java POJO that is contained by the parent object as a bean added with the addBean method.
```
Comment (gregw) [inline jetty-documentation/src/main/asciidoc/administration/logging/jetty-server-dump.adoc]: Does this list render OK. Should the key entries be quoted?

File: jetty-documentation/src/main/asciidoc/administration/logging/jetty-server-dump.adoc [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+- += managed is a bean that is also a LifeCycle that is started and stopped with the parent object.
+- +~ unmanaged is a bean that is also a LifeCycle that is started and stopped with the parent object. It is typically shared with other objects (hence its children are not dumped)
+- +? auto is a bean that has been added to an unstarted parent. If it is a LifeCycle that is not started when the parent is started, then it is started and becomes a managed bean, otherwise it becomes either unmanaged or just a bean
+- +: iterable is an object that is contained within an iterable field of the parent (eg a list, set etc.)
+- +] array is an object that is contained within an array field of the parent
+- +@ map' is an object that is contained within an map field of the parent +> undefined` is an object that is contained within the parent by an undefined relationship
+
+
+==== Examining a Jetty Distribution Dump
```
Comment (gregw) [inline jetty-documentation/src/main/asciidoc/administration/logging/jetty-server-dump.adoc]: Examining?   I think 'An Example' would be better

**Review by gregw — CHANGES_REQUESTED** [review]
A few changes.

But more importantly I think we need to chat with @sbordet regarding his reorg of the doco into programming and operational guides.     This section repeatedly has both programmatic configuration and parameter configuration.... so perhaps it should be split into two now, with most detail being in the programming version

File: jetty-documentation/src/main/asciidoc/administration/logging/jetty-server-dump.adoc (hunk 3 above)
Comment (lachlan-roberts) [reply jetty-documentation/src/main/asciidoc/administration/logging/jetty-server-dump.adoc]: The list renders fine, I have backquoted the titles to make them clearer.

**Review by gregw — APPROVED** [review]
This is definitely better than what we have.... but the merge to 10 will need to split the content to programming and operations (chat to @sbordet ).  So perhaps be ready with your jetty-10 changes before merging this.

**Pull request merged**